src/internal/storage/chunk/storage.go

```diff
@@ -81,3 +82,35 @@ func (s *Storage) List(ctx context.Context, cb func(id ID) error) error {
 func (s *Storage) NewDeleter() track.Deleter {
 	return &deleter{}
 }
+
+// Check runs an integrity check on the objects in object storage.
+// It will check objects for chunks with IDs in the range [first, last)
+// As a special case: if len(end) == 0 then it is ignored.
+func (s *Storage) Check(ctx context.Context, begin, end []byte, readChunks bool) (int, error) {
```

Comment:
It seems like this should expose checking one or all chunks rather than a range. What is the use of checking a range of chunks? Also, we can check past the end id with the current implementation.

Reply:
So that this work can be parallelized.   We can use the Task Service at the PFS level to make multiple calls to `Check`.  For now though, it provides a way to resume manually if something goes wrong.

As long as the batch size is reasonably small I don't think the overrun is that big of a deal.  It's not like it mutates anything.

Reply:
Okay, that makes sense. Does it still make sense to batch under the hood in that case? It seems like we should have a function that shards the chunks similar to the way we shard file sets and the workers would call this check function with those batches. Also it seems like it would make sense for the exported function to operate on chunk.IDs.

Reply:
For the sharding I was thinking we would generate distinct prefixes in a loop.  First `[]byte{0}` then `[]byte{1}` etc. all the way up to `[]byte{255}`.  In this case begin and end would not be valid chunk.IDs, since they are not long enough.  I guess we could also pad them with a suffix of zeros.  At some point we should change ID to `type ID = pachhash.Output`